src/pages/AdvancedSearch/ResultsBlock/ResultsBlock.tsx

```diff
@@ -0,0 +1,84 @@
+import {
+  CodeSnippet,
+  CodeSnippetBlockAppearance,
+  Spinner,
+} from "@canonical/react-components";
+import { useState } from "react";
+
+import {
+  getCrossModelQueryLoaded,
+  getCrossModelQueryLoading,
+  getCrossModelQueryResults,
+} from "store/juju/selectors";
+import { useAppSelector } from "store/store";
+
+enum CodeSnippetView {
+  TREE = "tree",
+  JSON = "json",
+}
+
+const ResultsBlock = (): JSX.Element | null => {
+  const isCrossModelQueryLoading = useAppSelector(getCrossModelQueryLoading);
+  const isCrossModelQueryLoaded = useAppSelector(getCrossModelQueryLoaded);
+  const crossModelQueryResults = useAppSelector(getCrossModelQueryResults);
+  const [codeSnippetView, setCodeSnippetView] = useState<CodeSnippetView>(
+    CodeSnippetView.TREE
+  );
+  const codeSnippetContent = {
+    tree: "",
+    json: JSON.stringify(crossModelQueryResults, null, 2),
+  };
+
+  if (isCrossModelQueryLoading) {
+    return (
+      <div className="u-align--center">
+        <div className="u-sv3">
+          <hr />
+        </div>
+        <Spinner />
+      </div>
+    );
+  }
```

Comment:
I changed the layout a little here. I was going to centre the spinner vertically in the empty space but that would have required a lot of other changes to the layout CSS. Maybe that can be done at the end if we have time.